index.js

```diff
@@ -363,7 +362,12 @@ var Base = Class.extend({
     }
 
     if (columnNameArray.length !== valueArray.length) {
-      return callback(new Error('The number of columns does not match the number of values.'));
+      var error = new Error('The number of columns does not match the number of values.');
+      if (typeof callback === 'function') {
+        return callback(error);
+      } else {
+        throw error;
```

Comment:
it is better to return Promise.reject(error), I will replace that afterwards myself, will merge this now.